sam/models/base_model.py

```diff
@@ -1,27 +1,29 @@
 import warnings
 from abc import ABC, abstractmethod
 from operator import itemgetter
-from typing import Any, Callable, Sequence, Tuple, Union, List
+from typing import Callable, List, Sequence, Tuple, Union
 
 import numpy as np
 import pandas as pd
+from sam.feature_engineering import BaseFeatureEngineer, IdentityFeatureEngineer
+from sam.models.utils import remove_target_nan, remove_until_first_value
 from sam.preprocessing import inverse_differenced_target, make_shifted_target
 from sam.utils import assert_contains_nans, make_df_monotonic
 from sklearn.base import BaseEstimator, RegressorMixin, TransformerMixin
-from sklearn.pipeline import Pipeline
 from sklearn.utils.validation import check_is_fitted
 
 
 class BaseTimeseriesRegressor(BaseEstimator, RegressorMixin, ABC):
     """
     This is an abstract class for all SAM models.
-    Every SAM model (including the most used SamQuantileMLP)
-    needs to inherit this class and implement the abstract methods.
+    Every SAM model (including the most used TimeseriesMLP) needs to inherit this class and
+    implement the abstract methods.
 
     There are some notes:
     - There is no validation yet. Therefore, the input data must already be sorted and monospaced
-    - The feature engineering is very simple, we just calculate lag/max/min/mean for a given window
-      size, as well as minute/hour/month/weekday if there is a time column
+    - The feature engineering should be provided as a any transformer. If the feature engineering
```

Comment:
Have we tested it also with a pipeline of transformers? Like combinig the SimpleFeatureEngineer with an imputer?

Reply:
I've added it as issue #27